docs/changelog/2065.md

```diff
@@ -0,0 +1,3 @@
+- Added header `precice/exceptions.hpp` which provides the preCICE exception type `precice::Error`. It is included via `precice/precice.hpp`.
+- Changed the C++ API to throw a `precice::Error` instead of aborting if preCICE detects an error. This allows adapters to clean up on error and makes preCICE easier to use via interactive sessions.
+- Fixed mismatched MPI communicator sizes to trigger an assertion before their sanitization check.
```

Comment:
Does this belong into this PR? Sounds to me a bit off-topic.

Reply:
I fixed this as part of the test that I wrote. Slightly off-topic yes.